Sources/Document/Document.swift

```diff
@@ -436,5 +515,76 @@ public actor Document {
         }
 
         self.defaultSubscribeCallback?(event)
+
+        if event.type == .peersChanged {
+            self.peersSubscribeCallback?(event)
+        }
+    }
+
+    private func isSameElementOrChildOf(_ elem: String, _ parent: String) -> Bool {
+        if parent == elem {
+            return true
+        }
+
+        let nodePath = elem.components(separatedBy: ".")
+        let targetPath = parent.components(separatedBy: ".")
+
+        var result = true
+
+        for (index, path) in targetPath.enumerated() where path != nodePath[safe: index] {
+            result = false
+        }
+
+        return result
+    }
+
+    /**
+     * `setOnlineClients` sets the given online client set.
+     */
+    func setOnlineClients(_ onlineClients: Set<ActorID>) {
+        self.onlineClients = onlineClients
+    }
+
+    /**
+     * `addOnlineClient` adds the given clientID into the online client set.
+     */
+    func addOnlineClient(_ clientID: ActorID) {
+        self.onlineClients.insert(clientID)
+    }
+
+    /**
+     * `removeOnlineClient` removes the clientID from the online client set.
+     */
+    func removeOnlineClient(_ clientID: ActorID) {
+        self.onlineClients.remove(clientID)
+    }
+
+    /**
+     * `hasPresence` returns whether the given clientID has a presence or not.
+     */
+    public func hasPresence(_ clientID: ActorID) -> Bool {
+        self.presences[clientID] != nil
+    }
+
+    /**
+     * `getPresence` returns the presence of the given clientID.
+     */
+    public func getPresence(_ clientID: ActorID) -> PresenceData? {
+        self.presences[clientID]
+    }
+
+    /**
+     * `getPresences` returns the presences of online clients.
+     */
+    public func getPresences() -> [PeerElement] {
+        var presences = [PeerElement]()
```

Comment:
[We decided to use the term `peer` to refer to other clients that do not include myself](https://github.com/yorkie-team/yorkie-js-sdk/issues/615). Therefore, it would be better to use other name instead of `PeerElement`.

This could also relate to the [yorkie#616](https://github.com/yorkie-team/yorkie/pull/616). (cc. @hackerwins )
One suggestion is to use the term `User` as the type.(refer to [liveblocks](https://github.com/liveblocks/liveblocks/blob/main/packages/liveblocks-core/src/types/User.ts)) This would refer to clients connected(watched) to the document. I've added commit that extracts the `User` type in [js-sdk#618](https://github.com/yorkie-team/yorkie-js-sdk/pull/618). 
We are currently considering the introduction of the `User` term as it did not exist before. I would like to know your thoughts on the `User` type and if you have any good ideas, please share them.

Reply:
I'll change the term "peer" to "user" later.